aecore_eqc/aec_dev_reward_eqc.erl

```diff
@@ -0,0 +1,80 @@
+-module(aec_dev_reward_eqc).
+
+-include_lib("eqc/include/eqc.hrl").
+
+-compile([export_all, nowarn_export_all]).
+
+prop_split() ->
+    ?FORALL(
+       {BeneficiaryReward1,
+        BeneficiaryReward2,
+        BeneficiariesMap,
+        UnallocatedShares},
+       {gen_beneficiary_reward(),
+        gen_beneficiary_reward(),
+        gen_beneficiaries_map(),
+        gen_unallocated_shares()},
+       begin
+           Beneficiaries = lists:sort(maps:to_list(BeneficiariesMap)),
+           AllocatedShares = allocated_shares(Beneficiaries),
+           TotalShares = AllocatedShares + UnallocatedShares,
+           {{AdjustedReward1,
+             AdjustedReward2},
+            DevRewards} =
+               aec_dev_reward:split_int(
+                 BeneficiaryReward1,
+                 BeneficiaryReward2,
+                 AllocatedShares,
+                 TotalShares,
+                 Beneficiaries),
+           is_reward(AdjustedReward1)
+               andalso is_reward(AdjustedReward2)
+               andalso (AdjustedReward1 =< BeneficiaryReward1)
+               andalso (AdjustedReward2 =< BeneficiaryReward2)
+               andalso (length(DevRewards) =:= length(Beneficiaries))
+               andalso (lists:sort(
+                          lists:map(fun({K, _}) -> K end, DevRewards))
+                        =:=
+                            lists:sort(
+                              lists:map(fun beneficiary_pubkey/1, Beneficiaries)))
+               andalso ([] =:= lists:filter(
+                                 fun({_, R}) -> not is_reward(R) end,
+                                 DevRewards))
+               andalso ((AdjustedReward1
+                         + AdjustedReward2
+                         + lists:sum(
+                             lists:map(fun({_, R}) -> R end, DevRewards)))
+                        =:=
+                            (BeneficiaryReward1
+                             + BeneficiaryReward2))
+       end).
```

Comment:
I did not enforce that beneficiaries with zero share have zero reward. I may.

Reply:
Makes any longer after https://github.com/aeternity/aeternity/pull/2343/commits/dbbbe814dacf24b16d4979db6c876c26785a70d0